homeassistant/components/homewizard/switch.py

```diff
@@ -0,0 +1,147 @@
+"""Creates Homewizard Energy switch entities."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import (
+    DEVICE_CLASS_OUTLET,
+    DEVICE_CLASS_SWITCH,
+    SwitchEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ENTITY_CATEGORY_CONFIG
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import COORDINATOR, DOMAIN, DeviceResponseEntry
+from .coordinator import HWEnergyDeviceUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switches."""
+
+    coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        COORDINATOR
+    ]
+
+    if coordinator.api.state:
+        async_add_entities(
+            [
+                HWEnergyMainSwitchEntity(coordinator, entry),
+                HWEnergySwitchLockEntity(coordinator, entry),
+            ]
+        )
+
+
+class HWEnergySwitchEntity(CoordinatorEntity[DeviceResponseEntry], SwitchEntity):
+    """Representation switchable entity."""
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        entry: ConfigEntry,
+        key: str,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator)
+        self.entry = entry
+
+        # Config attributes
+        self.api = coordinator.api
+        self._attr_unique_id = f"{entry.unique_id}_{key}"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return information about the device."""
+        return {
+            "name": self.entry.title,
+            "manufacturer": "HomeWizard",
+            "sw_version": self.data["device"].firmware_version,
+            "model": self.data["device"].product_type,
+            "identifiers": {(DOMAIN, self.data["device"].serial)},
+        }
+
+    @property
+    def data(self) -> DeviceResponseEntry:
+        """Return data from DataUpdateCoordinator."""
+        return self.coordinator.data
+
+
+class HWEnergyMainSwitchEntity(HWEnergySwitchEntity):
+    """Representation of the main power switch."""
+
+    def __init__(
+        self, coordinator: HWEnergyDeviceUpdateCoordinator, entry: ConfigEntry
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator, entry, "power_on")
+
+        # Config attributes
+        self._attr_name = entry.title
```

Comment:
Feels a bit weird, that if the entry title change, the entity name would change as well?

Reply:
usually `<device name> <entity name>` (if provided by the API).
Otherwise just an entity name `Switch`

Reply:
Added `Switch` name to entity c93f798